src/System.Data.Common/tests/System/Data/DataTableTest2.cs

```diff
@@ -1443,20 +1369,11 @@ public void Select_ByFilter()
             Assert.Equal(al.ToArray(), drSelect);
         }
 
-        private void CompareUnSorted(Array a, Array b)
+        private void CompareUnsorted<T>(T[] a, T[] b)
         {
-            string msg = string.Format("Failed while comparing(Array a ={0} ({1}), Array b = {2} ({3}))]", a.ToString(), a.GetType().FullName, b.ToString(), b.GetType().FullName);
-            foreach (object item in a)
-            {
-                if (Array.IndexOf(b, item) < 0) //b does not contain the current item.
-                    Assert.False(true);
-            }
-
-            foreach (object item in b)
-            {
-                if (Array.IndexOf(a, item) < 0) //a does not contain the current item.
-                    Assert.False(true);
-            }
+            IEnumerable<T> union = a.Union(b);
+            IEnumerable<T> intersection = a.Intersect(b);
+            Assert.Equal(union.Count(), intersection.Count());
```

Comment:
Not sure why I've done this... but hope it's correct at least 😅

Reply:
This doesn't take into account the same element appearing more than once on one side but not the other... but neither did the previous implementation... Otherwise seems OK, could inline and make into expression-bodied.